python/tvm/relay/op/contrib/cmsisnn.py

```diff
@@ -287,8 +287,8 @@ def check_qnn_avg_pool2d(pattern):
         return (
             pooling.attrs.layout == "NHWC"
             and int(input_op.checked_type.shape[0]) == 1
-            and input_op.checked_type.dtype == "int8"
-            and output.checked_type.dtype == "int8"
+            and (input_op.checked_type.dtype == "int8" or input_op.checked_type.dtype == "int16")
+            and (output.checked_type.dtype == "int8" or output.checked_type.dtype == "int16")
```

Comment:
Minor suggestion: a slightly more pythonic comparison.

```suggestion
            and (input_op.checked_type.dtype in ("int8", "int16"))
            and (output.checked_type.dtype in ("int8", "int16"))
```

Reply:
Just for completeness, there is a similar one in `qnn_max_pool2d_pattern`.